version: '2.1'

services:
  search-service-db-node-ex01:
    build:
      context: ./sources/search_service/mariadb
      dockerfile: Dockerfile
    container_name: search-service-db-node-ex01
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: archdb
      MYSQL_USER: admin
      MYSQL_PASSWORD: admin
      MYSQL_ROOT_PASSWORD: '1'
    command: >
      --sql-mode='STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'
      --general-log=ON
      --log-queries-not-using-indexes=ON
    networks:
      - arch-network
    ports:
      - 3360:3306
    healthcheck:
      test: "/usr/bin/mysql --user=stud --password=stud --execute \"SHOW DATABASES;\""
      timeout: 20s
      retries: 5
    volumes:
      - search-service-db-node-ex01-data:/var/lib/mysql

  search_service:
    build:
      context: sources/search_service
      dockerfile: Dockerfile
    container_name: search_service
    environment:
      - DB_HOST=search-service-db-node-ex01
      - DB_PORT=3306
      - DB_LOGIN=admin
      - DB_PASSWORD=admin
      - DB_DATABASE=archdb
    ports:
      - 8080:8080
    networks:
      - arch-network
    depends_on:
      search-service-db-node-ex01:
        condition: service_healthy

volumes:
  search-service-db-node-ex01-data:

networks:
  arch-network:
    driver: "bridge"
    external: false